public_html/wp-content/sunrise.php

```diff
@@ -3,154 +3,27 @@
- */
-if ( 'cli' === php_sapi_name() && defined( 'CLI_HOSTNAME_OVERRIDE' ) ) {
-	$_SERVER['HTTP_HOST'] = str_replace( 'wordcamp.org', CLI_HOSTNAME_OVERRIDE, $_SERVER['HTTP_HOST'] );
-}
-
-/*
- * This must be enabled for `get_corrected_root_relative_url()` to work for images. If we ever want to disable it,
- * we'll need to come up with another solution for that, (e.g., nginx rewrite rules).
- *
- * This needs to be applied outside of `main()` so that it takes effect in CLI environments, for consistency.
- */
-add_filter(
-	'pre_site_option_ms_files_rewriting',
-	function() {
-		return '1';
-	}
-);
-
-// Redirecting would interfere with bin scripts, unit tests, etc.
-if ( php_sapi_name() !== 'cli' ) {
-	main();
-}
+load_network_sunrise();
 
 
 /**
- * Preempt `ms_load_current_site_and_network()` in order to set the correct site.
+ * Load the sunrise file for the current network.
  */
-function main() {
-	list(
-		'domain' => $domain,
-		'path'   => $path
-	) = guess_requested_domain_path();
-
-	add_action( 'template_redirect', __NAMESPACE__ . '\redirect_duplicate_year_permalinks_to_post_slug' );
-
-	$status_code = 301;
-
-	$redirect = root_redirects( $domain, $_SERVER['REQUEST_URI'] );
-
-	if ( ! $redirect ) {
-		$redirect = domain_redirects( $domain, $path, $_SERVER['REQUEST_URI'] );
-	}
-
-	if ( ! $redirect ) {
-		$redirect = get_city_slash_year_url( $domain, $_SERVER['REQUEST_URI'] );
-	}
-
-	/*
-	 * This has to run before `get_canonical_year_url()`, because that function will redirect these requests to
-	 * the latest site instead of the intended one, and would strip out the request URI.
-	 *
-	 * This branch specifically only processes root-relative urls which are linked to from the same domain.
-	 */
-	if ( ! $redirect ) {
-		$redirect = get_corrected_root_relative_url( $domain, $path, $_SERVER['REQUEST_URI'], $_SERVER['HTTP_REFERER'] ?? '' );
-
-		if ( $redirect ) {
-			/*
-			 * This isn't a permanent redirect, because the value changes based on the referrer. `europe.wordcamp.org/2019` and
-			 * `europe.wordcamp.org/2020` might both have `/tickets` links that would both resolve to
-			 * `europe.wordcamp.org/tickets`, but the request will be routed to a different site each time, based on the referrer.
-			 */
-			$status_code = 302;
-		}
-	}
-
-	// Do this one last, because it sometimes executes a database query.
-	if ( ! $redirect ) {
-		$redirect = get_canonical_year_url( $domain, $path );
-
-		/*
-		 * Keep the path components if we requested the root site.
-		 * Eg. narnia.wordcamp.org/tickets => narnia.worcamp.org/{$latest_year}/tickets
-		 *     narnia.wordcamp.org/wp-login.php => narnia.wordcamp.org/{$latest_year}/wp-login.php
-		 */
-		if ( $redirect && '/' === $path ) {
-			$redirect .= ltrim( $_SERVER['REQUEST_URI'], '/' );
-		}
+function load_network_sunrise() {
+	$tld = get_top_level_domain();
 
-		if ( $redirect ) {
-			$status_code = 302;
-		}
-	}
+	switch ( DOMAIN_CURRENT_SITE ) {
```

Comment:
I'd have been tempted to just use `SITE_ID_CURRENT_SITE` here, the network ID, referring back to a constant..

```suggestion
	switch ( SITE_ID_CURRENT_SITE ) {
		case NETWORK_ID_EVENTS:
			require __DIR__ . '/sunrise-events.php';
			break;
		default:
		case NETWORK_ID_WORDCAMP:
			require __DIR__ . '/sunrise-wordcamp.php';
			break;
	}
```

Reply:
👍🏻